compiler/semantic/sql.go

```diff
@@ -629,17 +620,90 @@ func (a *analyzer) semJoinCond(cond ast.JoinExpr) (dag.Expr, dag.Expr, error) {
 		if !ok {
 			return e, e, nil
 		}
-		leftKey, rightKey := key, key
-		if a.scope.schema != nil {
-			leftKey = &dag.This{Kind: "This", Path: append([]string{"left"}, key.Path...)}
-			rightKey = &dag.This{Kind: "This", Path: append([]string{"right"}, key.Path...)}
-		}
-		return leftKey, rightKey, nil
+		return key, key, nil
 	default:
 		panic(fmt.Sprintf("semJoinCond: unknown type: %T", cond))
 	}
 }
 
+func (a *analyzer) semEquiJoin(leftAlias, rightAlias string, cond *ast.JoinOnExpr, lhs, rhs ast.Expr) (dag.Expr, dag.Expr) {
```

Comment:
Nits: The usual pattern in this package is to name a method after the thing it analyzes and to pass the thing as the first argument.

```suggestion
func (a *analyzer) semJoinOnExpr(cond *ast.JoinOnExpr, leftAlias, rightAlias string, lhs, rhs ast.Expr) (dag.Expr, dag.Expr) {
```